app/code/Magento/Cms/Ui/Component/DataProvider.php

```diff
@@ -101,21 +116,42 @@ public function prepareMetadata()
                                 'editorConfig' => [
                                     'enabled' => false
                                 ],
-                                'componentType' => \Magento\Ui\Component\Container::NAME
+                                'componentType' => Container::NAME
                             ]
                         ]
                     ]
                 ]
             ];
         }
 
+        if (!$this->getAuthorizationInstance()->isAllowed('Magento_Cms::save_design')) {
+
+            foreach ($this->pageLayoutColumns as $column) {
+                $metadata['cms_page_columns']['children'][$column] = [
+                    'arguments' => [
+                        'data' => [
+                            'config' => [
+                                'editor' => [
+                                    'editorType' => false
+                                ],
+                                'componentType' => Container::NAME
+                            ]
+                        ]
+                    ]
+                ];
+            }
+        }
+
         return $metadata;
     }
 
     /**
-     * @inheritdoc
+     * Add Filter
+     *
+     * @param Filter $filter
+     * @return void
```

Comment:
doesn't magento use any rector/phpcs? these docblock are redundant when there's proper typing

Reply:
It's their coding standards requiring it, there's been [a long time request](https://github.com/magento/magento-coding-standard/issues/404) to stop doing this [and a fix](https://github.com/magento/magento-coding-standard/pull/406), but it's been ignored for a long time :(

(also their entire REST API functionality depends on types defined in docblocks at the moment, instead of proper type definitions in code, so docblock types can't be removed everywhere just yet, but it can be removed in most classes)

Just FYI

Reply:
Thanks for this great info!